### DISCUSSION TITLE
Do transactions make multiple requests? Is it possible to make just one request?

### DISCUSSION THREAD

**Discussion by KernelDeimos** [sidorares/node-mysql2#2096] [Q&A]
Hi, I'm dealing with mysql servers with a high round-trip latency. To deal with the situation I've created a class that mediates all write operations (insert, update, delete) so that they can be queued and sent to the mysql server in one go.

The trouble is, when I use a transaction I don't get the performance improvement I was hoping for. The data suggests that each operation in the transaction is being sent to the mysql server in a separate request.

Is mysql2 really making multiple separate requests in sequence? Is there a way to not do that?

**Comment by sidorares** [comment]
There is no way of doing that, in mysql protocol you are only allowed to send next query after receiving results from a previous command. The only options for you are stored procedures ( if allowed by your server, for example Planetscale does not have SPs ) or sending multiple queries separated by ";" ( as you observed in https://github.com/sidorares/node-mysql2/issues/2097 you can't use prepared statements here but text protocol aka `.query()` should work )

**Reply by KernelDeimos** [reply]
I read somewhere about concerns with sql injection vulnerabilities if `multipleStatements` is enabled. Do these concerns still apply if I'm using parameter binding everywhere?

**Reply by sidorares** [reply]
Yes, it's a possible attack vector that you should pay attention but if the data that you don't control is escaped correctly should be of no issue
